calling-apis/managing-asynchronous-calls.md

```diff
@@ -0,0 +1,277 @@
+
+## Experiment: Making Multiple Calls
+
+Raffy is building a map web app. He wants to run an experiment!
+
+He wants to make two API requests to the [LocationIQ APIs](https://locationiq.com/docs):
+
+1. First, he wants to make a request to the Forward Geocoding API to find the latitude and longitude of "Seattle, Washington, USA."
+1. Then, he wants to make a request to the Reverse Geocoding API with the latitude and longitude he found, and see if the results come back as "Seattle, Washington, USA."
+
+He decides to run his experiment with [this code](https://replit.com/@adacore/Managing-Asynchronous-Requests)!
+
+<!-- prettier-ignore-start -->
+```js
+const axios = require("axios");
+
+const YOUR_API_KEY = 'YOUR API KEY';
+
+const findLatitudeAndLongitude = (query) => {
+  let latitude, longitude;
+  axios.get('https://us1.locationiq.com/v1/search.php',
+  {
+    params: {
+      key: YOUR_API_KEY,
+      q: 'Seattle, Washington, USA',
+      format: 'json'
+    }
+  })
+  .then( (response) => {
+    latitude = response.data[0].lat;
+    longitude = response.data[0].lon;
+    console.log('success in findLatitudeAndLongitude!', latitude, longitude);
+  })
+  .catch( (error) => {
+    console.log('error in findLatitudeAndLongitude!');
+  });
+
+  return {
+      seattleLat: latitude,
+      seattleLon: longitude
+  }
+}
+
+const findLocation = (latitude, longitude) => {
+  axios.get('https://us1.locationiq.com/v1/reverse.php',
+  {
+    params: {
+      key: YOUR_API_KEY,
+      format: 'json',
+      lat: latitude,
+      lon: longitude
+    }
+  })
+  .then( (response) => {
+    console.log('success in findLocation!', response.data);
+    return response.data;
+  })
+  .catch( (error) => {
+    console.log('error in findLocation!');
+  });
+}
+
+const seattleCoordinates = findLatitudeAndLongitude('Seattle, Washington, USA');
+
+const locations = findLocation(seattleCoordinates.seattleLat, seattleCoordinates.seattleLon);
+
+console.log(locations);
+```
+<!-- prettier-ignore-end -->
+
+This code:
+
+- Defines `findLatitudeAndLongitude`, which searches for latitude and longitude with the given query
+- Defines `findLocation`, which searches for location with the given latitude and longitude
+- Finds the coordinates of Seattle by calling `findLatitudeAndLongitude` and puts it in `seattleCoordinates`
+- Finds the location results by calling `findLocations` and puts it in `locations`
+
+However, when we run this code, our console outputs this:
+
+```
+error in findLocation!
+success in findLatitudeAndLongitude! 47.6038321 -122.3300624
+```
+
+Raffy's experiment doesn't look successful at all! What's noticeable is that `error in findLocation!` prints out **_before_** `success in findLatitudeAndLongitude!` does.
+
+It's apparent that the API call in `findLatitudeAndLongitude` works. We can get back a latitude and longitude just fine!
```

Comment:
I don't like to assume that it's apparent to a newbie that the function is successful 

```suggestion
Because `findLatitudeAndLongitude` prints a success message, it's apparent that the API call in `findLatitudeAndLongitude` works. We can get back a latitude and longitude just fine!
```

Reply:
I like Chris' suggestion about saying what makes it apparent, but since Raffy is still investigating, I'd soften saying that it's apparent.
Also, saying we get the values makes it sound like they're being returned fine, but really they're just being printed, so state that explicitly

 ```suggestion
Because `findLatitudeAndLongitude` prints a success message, it seems like the API call in `findLatitudeAndLongitude` works. The latitude and longitude values print just fine!
```